gnomad/assessment/sanity_checks.py

```diff
@@ -190,3 +212,744 @@ def compare_row_counts(ht1: hl.Table, ht2: hl.Table) -> bool:
+    verbose: bool,
+    show_percent_sites: bool = True,
+    delimiter: str = "-",
+    metric_first_label: bool = True,
+) -> None:
+    """
+    Perform sanity checks on frequency data in input Table.
+
+    Check:
+        - Number of sites where callset frequency is equal to a subset frequency (raw and adj)
+            - eg. t.info.AC-adj != t.info.AC-subset1-adj 
+        - Total number of sites where the allele count annotation is defined (raw and adj)
+        
+    :param t: Input MatrixTable or Table.
+    :param subsets: List of sample subsets.
+    :param verbose: If True, show top values of annotations being checked, including checks that pass; if False, show only top values of annotations that fail checks.
+    :param show_percent_sites: If true, show the percentage and count of overall sites that fail; if False, only show the number of sites that fail.
+    :param delimiter: String to use as delimiter when making group label combinations.
+    :param metric_first_label: If True, metric precedes label group, e.g. AC-afr-male. If False, label group precedes metric, afr-male-AC.
+    :return: None
+    """
+    t = t.rows() if isinstance(t, hl.MatrixTable) else t
+
+    field_check_expr = {}
+    for subset in subsets:
+        if subset:
+            subset += delimiter
+            for field in ["AC", "AN", "nhomalt"]:
+                for group in ["adj", "raw"]:
+                    logger.info(
+                        "Comparing subset %s frequencies to entire callset", subset
+                    )
+                    check_field_left = f"{field}{delimiter}{group}"
+                    if metric_first_label:
+                        check_field_right = f"{field}{delimiter}{subset}{group}"
+                    else:
+                        check_field_right = f"{subset}{field}{delimiter}{group}"
+
+                    field_check_expr[f"{check_field_left} != {check_field_right}"] = {
+                        "expr": hl.agg.count_where(
+                            t.info[check_field_left] == t.info[check_field_right]
+                        ),
+                        "display_fields": hl.struct(
+                            **{
+                                check_field_left: t.info[check_field_left],
+                                check_field_right: t.info[check_field_right],
+                            }
+                        ),
+                    }
+
+    generic_field_check_loop(
+        t, field_check_expr, verbose, show_percent_sites=show_percent_sites,
+    )
+
+    freq_counts = t.aggregate(
+        hl.struct(
+            total_defined_AC=hl.agg.count_where(
+                hl.is_defined(t.info[f"AC{delimiter}adj"])
+            ),
+            total_defined_AC_raw=hl.agg.count_where(
+                hl.is_defined(t.info[f"AC{delimiter}raw"])
+            ),
+        )
+    )
+    logger.info("Frequency spot check counts: %s", hl.eval(freq_counts))
+
+
+def sample_sum_sanity_checks(
```

Comment:
hmm, I tried using the "sanity" in the main functions but now I'm reconsidering that as since they will all live in a sanity check module. I think I need to think on this one a bit more but will propose alternatives. Maybe the method above could be sum_group_callstats or group_callstat_check?

Reply:
good point about the sanity check module haha. I like sum_group_callstats!